Rog/src/main/java/springbootMigracion/java/com/example/demo/controller/InstrumentoFinancieroController.java

```diff
@@ -0,0 +1,83 @@
+@Slf4j
+public class InstrumentoFinancieroController implements IInstrumentoFinancieroApi {
+
+    @Autowired
+    private IInstrumentoFinancieroService instrumentoFinancieroService;
+
+    @PostMapping("/registrar")
+    public ResponseEntity<?> registrarInstrumento(@RequestBody InstrumentoDTO instrumentoDTO) throws Exception {
+        log.info("Solicitud para agregar instrumento: {}", instrumentoDTO);
+        InstrumentoFinanciero instrumento = instrumentoFinancieroService.registrarInstrumento(instrumentoDTO);
+        log.info("Instrumento agregado exitosamente: {}", instrumento);
+        return ResponseEntity.status(HttpStatus.CREATED).body(instrumento);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<InstrumentoFinanciero>> listarInstrumentos() {
+        List<InstrumentoFinanciero> instrumentos = instrumentoFinancieroService.listarTodosLosInstrumentos();
+        log.info("Instrumentos listados: {}", instrumentos.size());
+        return ResponseEntity.ok(instrumentos);
+    }
+
+    @GetMapping("/buscar/{nombre}")
+    public ResponseEntity<List<InstrumentoFinanciero>> listarInstrumentosPorNombre(@PathVariable String nombre) {
+        List<InstrumentoFinanciero> instrumentosPorNombre = instrumentoFinancieroService.listarInstrumentosPorNombre(nombre);
+        log.info("Instrumentos encontrados por nombre: {}", instrumentosPorNombre.size());
+        return ResponseEntity.ok(instrumentosPorNombre);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<?> buscarInstrumentoPorId(@PathVariable Long id) throws Exception {
+        log.info("Solicitud para buscar instrumento por ID: {}", id);
+        InstrumentoFinanciero instrumento = instrumentoFinancieroService.buscarInstrumentoPorId(id);
+        log.info("Instrumento encontrado: {}", instrumento);
+        return ResponseEntity.ok(instrumento);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<?> eliminarInstrumento(@PathVariable Long id) throws Exception {
+        log.info("Solicitud para eliminar instrumento con ID: {}", id);
+        instrumentoFinancieroService.eliminarInstrumento(id);
+        log.info("Instrumento con ID {} eliminado", id);
+        return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<?> editarInstrumento(@PathVariable Long id, @RequestBody InstrumentoDTO nuevoInstrumentoDTO) throws Exception {
+        log.info("Solicitud para editar instrumento con ID: {}", id);
+        InstrumentoFinanciero instrumentoFinancieroActualizado = instrumentoFinancieroService.editarInstrumento(id, nuevoInstrumentoDTO);
+        log.info("Instrumento con ID {} actualizado exitosamente", id);
+        return ResponseEntity.ok(instrumentoFinancieroActualizado);
+    }
+
+    @PostMapping("/suscribir/{nombreInstrumento}/{nombreInversor}")
+    public ResponseEntity<?> suscribirInversor(@PathVariable String nombreInstrumento, @PathVariable String nombreInversor) throws Exception {
```

Comment:
Para que no existan dudas es necesario aclarar bien en la URL lo que vamos a registrar, al ser una dupla no veo problema sea por URL todo:

```suggestion
    @PostMapping("/suscribir/{nombreInstrumento}/inversor/{nombreInversor}")
    public ResponseEntity<?> suscribirInversor(@PathVariable String nombreInstrumento, @PathVariable String nombreInversor) throws Exception {
```